extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaConsumerConfigs.java

```diff
@@ -30,15 +30,23 @@
  */
 public class KafkaConsumerConfigs
 {
-
-  public static Map<String, Object> getConsumerProperties()
+  public static Map<String, Object> getConsumerProperties(Map<String, Object> customerConsumerProperties)
   {
+    final Object value = customerConsumerProperties.remove("consumeTransactionally");
```

Comment:
Putting this property in `KafkaIndexTaskIOConfig.consumerProperties` might cause some confusion. `consumerProperties` is designed to store kafka official properties. It's better to put the new property in `KafkaIndexTaskIOConfig` directly and put the code related to this property in `KafkaRecordSupplier.addConsumerPropertiesFromConfig`

Reply:
Hi FrankChen021, I have put the new property `consumeTransactionally` in the `KafkaIndexTaskIOConfig` directly. If users ignore this config or set this config true, Druid would consume Kafka transactionally same as current Druid behavior .

Set consumeTransactionally false here can disable Druid to consume Kafka transactionally, meanwhile Druid could consume lower version of Kafka now like 0.10.2.1 .

In this way, we don't change current Druid default behavior and provide a way to let Druid consume lower version Kafka.

I have tested this PR in our Dev Druid cluster.  Please take a look.